src/gmp_backend.rs

```diff
@@ -223,11 +237,20 @@ impl Bn {
 
     /// Convert this big number to a big-endian byte sequence
     pub fn to_bytes(&self) -> Vec<u8> {
+        debug_assert!(self >= &Self::zero());
+
+        if self < &Self::zero() {
+            return vec![0];
+        }
+
         let mut s = self.0.to_str_radix(16);
+
         if s.len() & 1 == 1 {
```

Comment:
I know this isn't your code, but this `if` statement could benefit from an explanation.  Perhaps we can mention it in an upstream PR.

Reply:
Sure